wasi/wasi.go

```diff
@@ -195,7 +195,25 @@ type API interface {
 	// TODO: PathSymlink
 	// TODO: PathUnlinkFile
 	// TODO: PollOneoff
-	// TODO: ProcExit
+
+	// ProcExit is the WASI function that terminates the execution of the module with an exit code.
+	// An exit code of 0 indicates successful termination. The meanings of other values are not defined by WASI.
+	//
+	// * rval - The exit code.
+	//
+	// In wazero, if ProcExit is called, store.CallFunction returns immediately, returning the given exit code as the error.
```

Comment:
oops this will drift I guess we can say "the calling function returns immediately" instead of linking X returnval parity